darwin.sh

```diff
@@ -1,61 +1,86 @@
+if ( which brew > /dev/null )
+then 
+  echo -e "\n==== Nano installed ====\n"
+else
+  echo -e "\n==== Installing Nano ====\n"
+  brew install nano
+fi
+
+# Install Git
+if ( which git > /dev/null )
+then 
+  echo -e "\n==== Git installed ====\n"
+else
+  echo -e "\n==== Installing Git ====\n"
+  brew install git
+fi
 
-# check operating system
-if [ "$(uname)" = 'Darwin' ]
+# Install Multipass
+if ( which multipass > /dev/null )
+then 
+  echo -e "\n==== Multipass installed ====\n"
+else
+  echo -e "\n==== Installing Multipass ====\n"
+  brew install --cask multipass
+fi
+
+# Set Multipass local driver to Qemu
+if [ "$(multipass get local.driver)" = "qemu" ]
 then
-  
-  echo "checking if brew is installed"  
-  if (which brew)                   
-  then 
-    echo "brew is installed"
-  else
-    echo "installing brew"
-    NONINTERACTIVE=1 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
-  fi
-
-  echo "checking if nano is installed"
-  if (which nano)
-  then 
-    echo "nano is already installed"
-  else
-    echo "installing nano"
-    brew install nano
-  fi
-
-  echo "checking if git is installed"
-  if (which git)
-  then
-    echo "git is already installed"
-  else
-    echo "installing git"
-    brew install git
-  fi
-
-  echo  "checking if multipass installed"
-  if (which multipass) 
-  then 
-    echo "multipass already installed" 
-  else
-    echo "installing multipass"
-    brew install --cask multipass
-  fi
+  echo -e -e "\n==== Qemu local driver set ====\n"
+else
+  echo -e -e "\n==== Setting Qemu as Multipass local driver ====\n"
+  multipass set local.driver=qemu
+fi
  
+# Create SSH keys
+if [ -f id_ed25519 ]
+then 
+  echo -e "\n==== SSH keys present ====\n"
 else
-  echo "This script is only for Mac. You are using $(uname). Please run the script for $(uname)."
+  echo -e "\n==== Creating SSH keys ====\n"
+  ssh-keygen -t ed25519 -f "id_ed25519" -N ''
 fi
 
-if [ -f ./id_ed25519 ] 
+# Create cloud-init.yaml
+if [ -f cloud-init.yaml ] 
 then
-  echo "ssh key pair already exists"
+  echo -e "\n==== Cloud-init.yaml present ====\n"
 else
-  echo "creating ssh key pair" 
-  ssh-keygen -t ed25519 -f "./id_ed25519" -N ''
+  echo -e "\n==== Creating cloud-init.yaml ====\n"
+  cat <<- EOF > cloud-init.yaml
+users:
+  - default
+  - name: $USER
+    sudo: ALL=(ALL) NOPASSWD:ALL
+    shell: /bin/bash
+    ssh_authorized_keys:
+      - $(cat id_ed25519.pub)
+EOF
+fi
+
+# Launch Relativepath instance
+if ( multipass list | grep relativepath | grep Running > /dev/null )
+then 
+  echo -e "\n==== Relativepath VM present ====\n"
+else 
+  echo -e "\n==== Creating Relativepath VM ====\n"
+  multipass launch --name relativepath --cloud-init cloud-init.yaml
 fi
 
-echo "launching multipass vm"
-multipass launch --name relativepath
+# SSH into Relativepath VM
+ssh -o StrictHostKeyChecking=no -i "./id_ed25519" $USER@$(multipass info relativepath | grep IPv4 | awk '{ print $2 }')
```

Comment:
This must be the alternative to the IP address. sweet!

Reply:
This IS the IP address, here we are grepping the IP (IP version 4) from the output of multipass info relativepath and capturing the IP address in a subshell. Pretty dope!

Reply:
Yes it is!